apps/golden-sample-app/src/app/app.error-handler.ts

```diff
@@ -0,0 +1,46 @@
+import { ErrorHandler, Injectable, NgZone } from '@angular/core';
+import { Router } from '@angular/router';
+
+function toErrorObject(error: Error | unknown): Error | undefined {
+  if (error && typeof error === 'object' && 'rejection' in error) {
+    error = (error as { rejection: Error | unknown }).rejection;
+  }
+
+  if (
+    error &&
+    typeof error === 'object' &&
+    'name' in error &&
+    'message' in error
+  ) {
+    return error as Error;
+  }
+
+  return undefined;
+}
+
+@Injectable()
+export class AppErrorHandler implements ErrorHandler {
+  constructor(
+    private readonly router: Router,
+    private readonly ngZone: NgZone
+  ) {}
+
+  handleError(error: Error | unknown | { rejection: Error | unknown }): void {
+    const errorObject = toErrorObject(error);
+
+    if (errorObject?.name === 'NotFoundError') {
+      this.ngZone.run(() => {
+        this.router.navigate(['/error'], {
```

Comment:
What do you think about this way of handling page errors which should be rendered as an error page?

Reply:
I think it's a nice example but probably not very feasible at scale as it implies having all error UIs in a central place.
We can leave it like this, but I'd prefer if we just handle the 404 route with the https://designsystem.backbase.com/v2/web-components/common-error-state/web component and delegate all the rest of the error handling isolating it in the journey libs. Its a pattern that scales better with the size of the team we have.